modules/videoio/cmake/detect_msdk.cmake

```diff
@@ -28,16 +28,24 @@ if(NOT HAVE_MFX)
 endif()
 
 if(HAVE_MFX AND UNIX)
-  find_path(MFX_va_INCLUDE va/va.h PATHS ${paths} PATH_SUFFIXES "include")
-  find_library(MFX_va_LIBRARY va PATHS ${paths} PATH_SUFFIXES "lib64" "lib/lin_x64")
-  find_library(MFX_va_drm_LIBRARY va-drm PATHS ${paths} PATH_SUFFIXES "lib64" "lib/lin_x64")
-  if(MFX_va_INCLUDE AND MFX_va_LIBRARY AND MFX_va_drm_LIBRARY)
-    list(APPEND MFX_INCLUDE_DIRS "${MFX_va_INCLUDE}")
-    list(APPEND MFX_LIBRARIES "${MFX_va_LIBRARY}" "${MFX_va_drm_LIBRARY}")
-    # list(APPEND MFX_LIBRARIES "-Wl,--exclude-libs=libmfx")
-  else()
+  foreach(mode NO_DEFAULT_PATH "")
+    find_path(MFX_va_INCLUDE va/va.h PATHS ${paths} PATH_SUFFIXES "include" ${mode})
+    find_library(MFX_va_LIBRARY va PATHS ${paths} PATH_SUFFIXES "lib64" "lib/lin_x64" ${mode})
+    find_library(MFX_va_drm_LIBRARY va-drm PATHS ${paths} PATH_SUFFIXES "lib64" "lib/lin_x64" ${mode})
```

Comment:
Because system _libva_ is not compatible with MediaSDK on older OSes (Ubuntu 16). It is possible to override _libva_ by setting all three cmake variables: `MFX_va_INCLUDE`, `MFX_va_LIBRARY`, `MFX_va_drm_LIBRARY`.